batch7crud-roll3/src/main/java/com/lftechnology/batch7crud/controller/StudentController.java

```diff
@@ -0,0 +1,181 @@
+package com.lftechnology.batch7crud.controller;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map.Entry;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.lftechnology.batch7crud.entity.Student;
+import com.lftechnology.batch7crud.exception.DataException;
+import com.lftechnology.batch7crud.service.StudentService;
+
+/**
+ * Servlet implementation class StudentController
+ */
+@WebServlet("/student/*")
+public class StudentController extends HttpServlet {
+	private static final long serialVersionUID = 1L;
+
+	protected void doGet(HttpServletRequest request, HttpServletResponse response)
+			throws ServletException, IOException {
+		// TODO Auto-generated method stub
+		String pathInfo = request.getPathInfo();
+		if (pathInfo == null) {
+			list(request, response, 1);
+		} else {
+			String[] pathParts = pathInfo.split("/");
+
+			if (pathParts[1].equals("create")) {
+				create(request, response);
+			} else if (pathParts[1] != null && pathParts.length == 2) {
+				int page = 1;
+				if (pathParts[1] != null) {
+					page = Integer.parseInt(pathParts[1]);
+				}
+				list(request, response, page);
+			} else if (pathParts[1] != null && pathParts[2].equals("edit")) {
+				edit(request, response, Integer.parseInt(pathParts[1]));
+
+			} else if (pathParts[1] != null && pathParts[2].equals("delete")) {
+				deleteProcess(request, response, Integer.parseInt(pathParts[1]));
+
+			} else {
+				response.sendRedirect("/batch7crud-roll3/student");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "/batch7crud-roll3/student" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)